presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -550,13 +550,14 @@ private void processProbe(LookupSource lookupSource)
                             break;
                         }
                     }
+                    // reset row join state for next row
+                    currentProbePositionProducedRow = false;
+                    statisticsCounter.recordProbe(joinSourcePositions);
+                    joinSourcePositions = 0;
                 }
-                currentProbePositionProducedRow = false;
```

Comment:
It means `currentProbePositionProducedRow` is no longer explicitly reset when we move to the next position. 
I think `currentProbePositionProducedRow` should be reset here or `advanceProbePosition` (equally non-ideal options), but rather not in the `if` above.

Also, i think the new code is not correct -- in case of spill if it affects current row, `currentProbePositionProducedRow` will remembered from last row being processed when we re-add the page.

Reply:
> It means currentProbePositionProducedRow is no longer explicitly reset when we move to the next position.

There is no point of that since only non -1 row change row state (`if (probe.getPosition() >= 0) {` condition).

I wanted to make row state reset consistent (after actual is processed), but also to avoid updating stats for -1 row.

> in case of spill if it affects current row, currentProbePositionProducedRow will remembered from last row being processed when we re-add the page.

When we add new probe, it's position is set to -1. When we restore row, we explicitly restore row state.